include/seqan3/range/views/persist.hpp

```diff
@@ -94,15 +85,18 @@ class view_persist : public std::ranges::view_interface<view_persist<urng_t>>
      *
      * No-throw guarantee.
      */
-    const_iterator begin() const noexcept
+    auto begin() noexcept
     {
         return std::ranges::begin(*urange);
     }
 
     //!\copydoc begin()
-    const_iterator cbegin() const noexcept
+    auto begin() const noexcept
+    //!\cond
+        requires const_iterable_range<urng_t>
+    //!\endcond
     {
-        return begin();
+        return std::ranges::cbegin(*urange);
```

Comment:
@rrahn, @smehringer, @eseiler This was a nice side effect of enforcing this rule, because `urange` is a pointer in this case and the const overload only has the effect that `urange` is now const (i.e. the pointer), but it is only shallow const, because it points to a non-const object `*urange`. This means `std::views::persist` never propagated the constness of the underlying range :)

Reply:
But this was right, since the persist view has pointer semantic similar to the ref_view of the standard. It does not modify the underlying range it only places it on the heap so that it allows to pipe rvalue non-view ranges. So it feels like it should be closer to the ref_view than to other views. But that is only a quick thought on this. Maybe we should discuss this.

Reply:
Core-Meeting:

we'll keep deep-constness (the current solution)